src/internal/coredb/projects.go

```diff
@@ -14,35 +14,81 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/pfs"
 )
 
+const (
+	pageSize = 100
+)
+
+// ProjectIterator batches a page of projectRow entries. Entries can be retrieved using iter.Next().
 type ProjectIterator struct {
-	*sqlx.Rows
+	pageNum int
```

Comment:
We probably want to guard this with a mutex. Imagine multiple goroutines iterating over this. However, maybe we should discourage that, but there's nothing we can do in code to enforce that without a lock.